Enhance AI financial advisor functionality with improved transaction …

sanjanarp opened the pull request:
…analysis and personalized advice prompts

sanjanarp committed:
Add Receipt Parser feature and enhance Bill Reminders functionality

sanjanarp committed:
Enhance AI financial advisor functionality with improved transaction analysis and personalized advice prompts

sanjanarp committed:
Merge branch 'Feature/BillReminder_ReceiptParser' into Feature/openai-enhancement

sanjanarp committed:
Enhance Receipt Parser functionality to support multiple file uploads and add transaction categorization

Buffden committed:
updated openai syntax as per the version>=1.0.0, added a script file to install dependencies and startup the backend and frontend with single command

Buffden committed:
Feature/complate-refactoring-receipt

Updated the whole UI/UX of receipt scanner,
added functionality to add multiple receipts by uploading and scanning,
Transactions are reflecting in insights page and budget tracker page as well,
Receipts are part of bank accounts selector as well, user can check or uncheck the receipts for it to be part of analysis or advises
refactored the logics of budget tracker page,
Updated how transaction tables are sorted

Buffden commented on frontend/streamlit_app.py:
This "if" condition manipulation was the reason for the bug: (application route not successfully landing on to "Add Bank Account" page after the addition of banks through Plaid.

If this code was for some other specific reason, please tell me.

Buffden committed:
Feature/plaid-re-routing-issue-fixed

fixed issue related to re route of the application  after adding bank accounts back to our "Add Bank Account" page
